### PR TITLE
cmd/geth: make console tests more robust

### PR DESCRIPTION
**Pull request by fjl**
This PR fixes some regressions introduced by #2535.
Please review: @karalabe 

Included in 1.4.6 milestone because it also fixes a hang in the `geth js` command 
(first commit, see #2651 for details).

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Thu Jun 2 20:54:54 UTC 2016_

File: console/console.go [hunk 1]
```diff
@@ -260,9 +260,9 @@ func (c *Console) Welcome() {
 			modules = append(modules, fmt.Sprintf("%s:%s", api, version))
 		}
 		sort.Strings(modules)
-		c.jsre.Run("(function () { console.log(' modules: " + strings.Join(modules, " ") + "'); })();")
+		fmt.Fprintln(c.printer, " modules:", strings.Join(modules, " "))
```
Comment (karalabe) [inline console/console.go]: Any particular reason for converting these to fmt, but not the welcome message part?

**Comment by karalabe** [comment]
Could we also add a test to geth that checks this particular scenario (i.e. `geth js empty.js`)? Would be nice to have this covered for next time :)

**Comment by karalabe** [comment]
Otherwise LGTM

**Comment by LefterisJP** [comment]
Works fine with a dummy.js file. :+1:

File: console/console.go (hunk 1 above)
Comment (fjl) [inline console/console.go]: The welcome message actually does requests and the code would be much longer if it were in Go.

Comment (fjl) [inline console/console.go]: fixed :P

**Comment by fjl** [comment]
@karalabe ok to merge after the changes?

File: console/console.go (hunk 1 above)
Comment (karalabe) [inline console/console.go]: Ok, lol :P

**Comment by karalabe** [comment]
:+1:

**Comment by fjl** [comment]
Waiting for CI.

**Pull request merged**